packages/appstore/src/components/modals/modal-manager.tsx

```diff
@@ -296,12 +295,32 @@ const ModalManager = () => {
         is_mt5_password_invalid_format_modal_visible ||
         is_sent_email_modal_enabled;
 
+    const is_invalid_investor_token =
+        Object.keys(current_list).length === 0 && localStorage.getItem('cfd_reset_password_code');
+    const should_show_cfd_reset_password_modal = is_cfd_reset_password_modal_enabled && !is_invalid_investor_token;
+
     return (
         <React.Fragment>
             {is_jurisdiction_modal_visible && <JurisdictionModal openPasswordModal={openRealPasswordModal} />}
             {should_show_cfd_password_modal && <CFDPasswordModal platform={platform} />}
             {is_cfd_verification_modal_visible && <CFDDbviOnBoarding />}
-            <CFDResetPasswordModal platform={platform} /> {/* a new condition for this hotfix needs to be found */}
+            {/* 
+                This modal used for reset password and always needs to be rendered!
+                OR
+                We can try to use the same logic from the modal for lazy loading like this: 
+
+                `
+                const is_invalid_investor_token =
+                    Object.keys(current_list).length === 0 && localStorage.getItem('cfd_reset_password_code');
+                const should_show_cfd_reset_password_modal = is_cfd_reset_password_modal_enabled && !is_invalid_investor_token;
+
+                in JSX:
+                {should_show_cfd_reset_password_modal && <CFDResetPasswordModal platform={platform} />}
+                `
+
+                !!! Needs to confirm with CFD squad !!!
```

Comment:
Hi @sergei-deriv 
Has this beed discussed with CFD Squad?

cc: @amina-deriv

Reply:
Hi @habib-deriv. Forgot to mention this team in `announce_reviews_fe` channel. Just now wrote to @shontzu-deriv about this. Waiiting for her response

Reply:
Nice, once done can proceed to delete to comments

Reply:
hi @sergei-deriv , sorry

this is LGTM, however, are you going to remove `is_open` from the modal itself? the code for the conditions are the same

![image](https://github.com/binary-com/deriv-app/assets/108507236/3ec59a81-bc9b-46d6-b2bf-ab7c1c7cb559)

Reply:
or pass is_open as a prop